### DISCUSSION TITLE
different output dimension of different scenarios with the same ss.TDS.config.tf=10

### DISCUSSION THREAD

**Discussion by xiaojiangsi** [CURENT/andes#429] [Q&A]
I used andes loop to generate data for time domain simulation under different faults and got different dimensional results, if I follow 0.033 steps it should be 308*27, but the final data appears in multiple cases. Why did this happen?
 for example :
![image](https://user-images.githubusercontent.com/51038918/226092826-bb665ddb-472b-41b0-89b7-54f768550a20.png)
 it cleared the fault at time 1.2, and the picture shows the part of the time series.
but when load level changed, the time series is as follow:
![image](https://user-images.githubusercontent.com/51038918/226092920-8180c360-cebf-4009-8cae-8ae97f08b856.png)
and somehow the dimension of the output is bigger than 308*27.
![image](https://user-images.githubusercontent.com/51038918/226092985-1aa5d2f0-a1f6-4a59-97a0-b42b5f30820e.png)

**Comment by cuihantao** [comment]
The trapezoidal method in ANDES uses a large step size by default. If nonconvergence for a step happpens, it will heuristically reduce the step size. When convergence improves, it will adjust the step size back to `TDS.config.tstep`. You can turn off the automatic shrinking of step size (see docs for `TDS.config.shrinkt`), but your first case will likely fail after t=1.0001.

If you need uniform step size, one way is to post-process the data and interpolate at the desired time. 

The second solution is to use a small step size (1/120, e.g.) and turn off `shrinkt`. This will slow down simulations across the board. Most cases should be able to complete, but there is no guarantee of convergence for all cases.

**Comment by jinningwang** [comment]
Hello @xiaojiangsi ,

Hope you find a helpful answer here. Please ***MARK OUT*** an answer to your question if you find any reply appropriate. In this way, the Q&A will be a valuable input to this community that can be helpful for future users.
We commit efforts to maintain this open-source software. We will be grateful if you can contribute to it if our work really helps you somewhere.

Regards,
Jinning
